features/rest/mapper/src/main/java/org/opennms/web/rest/mapper/v2/AlarmMapper.java

```diff
@@ -89,8 +88,8 @@ protected void fillAlarm(OnmsAlarm alarm, @MappingTarget AlarmDTO alarmDTO) {
         if (alarm.getTTicketId() != null && !alarm.getTTicketId().isEmpty() && ticketUrlTemplate != null) {
             alarmDTO.setTroubleTicketLink(getTicketUrl(alarm.getTTicketId()));
         }
-        if (alarm instanceof Situation) {
-            alarmDTO.setRelatedAlarms(((Situation)alarm).getAlarms().stream()
+        if (alarm.isSituation()) {
```

Comment:
Is there any need for this explicit check here? If it's not a situation then `getRelatedAlarms()` must return an empty collection.

Reply:
minimally. it only matters if the serialization does not include null fields and then the empty list  is not sent over the wire.
I will remove.